### PR TITLE
mod+GitHub: shorten PR template, bump golang to 1.15

### PR DESCRIPTION
**Pull request by guggero**
The golang version always needs to be in sync between the go.mod, the PR
checklist and the installation instructions.

We also add a new entry to the PR checklist for the recently added
release notes check.

#### Pull Request Checklist

- [ ] All changes are Go version 1.15 compliant
- [ ] Your PR passes all CI checks
- [ ] If this is your first time contributing, we recommend you read the [Code
  Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md)
   - [ ] The code being submitted is commented according to [Code Documentation and
  Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
   - [ ] Commits have a logical structure according to [Ideal Git Commit
  Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)
- [ ] For new code: Code is accompanied by tests which exercise both
  the positive and negative (error paths) conditions (if applicable)
- [ ] For bug fixes: Code is accompanied by new tests which trigger the bug
  being fixed to prevent regressions
- [ ] Any new logging statements use an appropriate subsystem and logging level
- [ ] For code and documentation: lines are wrapped at 80 characters
  (the tab character should be counted as 8 characters, not 4, as some IDEs do
  per default)
- [ ] All commits build properly and pass tests. Only in exceptional
  cases it can be justifiable to violate this condition. In that case, the
  reason should be stated in the commit message.

### CODE REVIEW

**Review by carlaKC — COMMENTED** [review]
LGTM! 

Unrelated to this change, but our PR checklist is pretty long now. One more doesn't hurt at this point, but since we have checks for a lot of these we could look into reducing a few of these bullet points to "actions pass"

**Review by carlaKC — APPROVED** [review]

**Comment by guggero** [comment]
> Unrelated(ish) to this change, but our PR checklist is pretty long now. One more doesn't hurt at this point, but since we have checks for a lot of these we could look into reducing a few of these bullet points to "actions pass"

I like that! Looks much better that way.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
   the positive and negative (error paths) conditions (if applicable)
-- [ ] For bug fixes: Code is accompanied by new tests which trigger
-  the bug being fixed to prevent regressions
-- [ ] Any new logging statements use an appropriate subsystem and
-  logging level
-- [ ] Code has been formatted with `go fmt`
-- [ ] Protobuf files (`lnrpc/**/*.proto`) have been formatted with
-  `make rpc-format` and compiled with `make rpc`
-- [ ] New configuration flags have been added to `sample-lnd.conf`
+- [ ] For bug fixes: Code is accompanied by new tests which trigger the bug
```
Comment (bhandras) [inline .github/pull_request_template.md:13]: nit: I'd rephrase this to "If possible code is accompanied...", since in some cases when code changes a lot it's impossible to create a regression test (for example crash fixes, data races etc.)

**Review by bhandras — APPROVED** [review]
LGTM :100:

File: .github/pull_request_template.md (hunk 1 above)
Comment (guggero) [reply .github/pull_request_template.md]: Makes sense, updated.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,27 +1,21 @@
...
 - [ ] For code and documentation: lines are wrapped at 80 characters
   (the tab character should be counted as 8 characters, not 4, as some IDEs do
   per default)
-- [ ] Running `make check` does not fail any tests
-- [ ] Running `go vet` does not report any issues
-- [ ] Running `make lint` does not report any **new** issues that did not
-  already exist
 - [ ] All commits build properly and pass tests. Only in exceptional
   cases it can be justifiable to violate this condition. In that case, the
   reason should be stated in the commit message.
```
Comment (carlaKC) [inline .github/pull_request_template.md:21]: Could this be collapsed into "Your PR passes all CI checks"? Now that we have the commit check, we don't really make this exception anymore?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,27 +1,21 @@
...
-- [ ] All changes are Go version 1.12 compliant
-- [ ] The code being submitted is commented according to [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
+   - [ ] The code being submitted is commented according to [Code Documentation and
+  Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
+   - [ ] Commits have a logical structure according to [Ideal Git Commit
+  Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)
 - [ ] For new code: Code is accompanied by tests which exercise both
   the positive and negative (error paths) conditions (if applicable)
-- [ ] For bug fixes: Code is accompanied by new tests which trigger
+- [ ] For bug fixes: If possible code is accompanied by new tests which trigger
```
Comment (carlaKC) [inline .github/pull_request_template.md:13]: nitnit: "If possible, code"

**Review by carlaKC — APPROVED** [review]
😍 looks so good! 

Only real comment is that the PR template looks _a little_ funny wrapped at 80 but that's  personal pref.

File: .github/pull_request_template.md (hunk 2 above)
Comment (guggero) [reply .github/pull_request_template.md]: Yes, good idea. What do you think about the current phrasing?

File: .github/pull_request_template.md (hunk 3 above)
Comment (guggero) [reply .github/pull_request_template.md]: Ah, wasn't sure here. Never got the hang of all the English comma rules :)

**Comment by guggero** [comment]
> Only real comment is that the PR template looks a little funny wrapped at 80 but that's personal pref.

You're right. Same with the release notes... Removed the line wrapping, hoping that nobody points out that the document itself violates the rule it imposes :)

**Comment by carlaKC** [comment]
> You're right. Same with the release notes... Removed the line wrapping, hoping that nobody points out that the document itself violates the rule it imposes :)

do as we say, not as we do ...😬

File: .github/pull_request_template.md (hunk 2 above)
Comment (carlaKC) [reply .github/pull_request_template.md]: Yeah that looks good!

File: .github/pull_request_template.md (hunk 3 above)
Comment (carlaKC) [reply .github/pull_request_template.md]: I _thiiink_ it's right, but tbh me too.
